tests/pytest_framework/src/syslog_ng/syslog_ng.py

```diff
@@ -27,22 +27,26 @@
 
 class SyslogNg(object):
     def __init__(self, logger_factory, instance_paths, syslog_ng_ctl):
+        self.instance_paths = instance_paths
         self.__syslog_ng_cli = SyslogNgCli(
             logger_factory, instance_paths, ConsoleLogReader(logger_factory, instance_paths), syslog_ng_ctl
         )
 
-    def start(self, config):
-        self.__syslog_ng_cli.start(config)
+    def start(self, config, external_tool=None):
```

Comment:
Back to the decorator-ish idea: what if the external tool were stored inside tc? Probably we could decorate tc to store it. Then this variant would be available for the constructors, and passed silently through the call chain without changing the user api at all (modulo the `@valgrind` decorator of course).

Reply:
+1, like the idea